mmv1/templates/terraform/nested_property_documentation.html.markdown.tmpl

```diff
@@ -1,10 +1,10 @@
 {{ "" }}
-{{- if $.FlattenObject }}
+{{- if and $.FlattenObject (not $.WriteOnlyProperties) }}
   {{- range $np := $.NestedProperties }}
 {{- trimTemplate "nested_property_documentation.html.markdown.tmpl" $np -}}
   {{- end -}}
-{{- else if $.NestedProperties }}
-<a name="nested_{{$.LineageAsSnakeCase}}"></a>The `{{ underscore $.Name }}` block {{ if $.Output }}contains{{ else }}supports{{ end }}:
```

Comment:
The change from `$.LineageAsSnakeCase` to `underscore $.Name` breaks many block links where those values are different. What was the intent of the change?

If we are changing this line we may want to break it into a dedicated change- that'll be easier to review, and will cut down tremendously on the changed files here.

Reply:
There must've been a change since opening the PR to now because I couldn't find any reason to do this. 

The original PR that adds the [initial changes to the MMv1 generation can be found here.](https://github.com/GoogleCloudPlatform/magic-modules/pull/12550/files#diff-0ee44b6dfb99752bf6dae3214ef04057cc0443041b6671cc6a1e3eb3355b9403) So it looks like it was changed but after that PR was merged. I've updated this here: https://github.com/GoogleCloudPlatform/magic-modules/pull/13135/commits/e7535e2d604a492bb33f8bade7013e445fbab02e